app/controllers/bookings_controller.rb

```diff
@@ -4,12 +4,26 @@ def new
     @booking = Booking.new
   end
 
-  # Create
-  # 1. Fetch the booked_date from the form params
-  # 2. Fetch the kondo_id from params[:kondo_id]
-  # 3. Fetch the user_id from current_user
-  # 4. Create the new booking with those three parameters and
-  # 5. Try to save
-  # 6. If saved -> redirect
-  # 7. If not saved -> render:new
+  def create
+    booked_date = date_formating(params[:booking])
+    @booking = Booking.new(user_id: current_user.id, kondo_id: params[:kondo_id], booked_date: booked_date)
+
+    if @booking.save
+      redirect_to kondo_bookings_path # Index of all bookings
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def date_formating(hash)
+    DateTime.new(
+      hash["booked_date(1i)"].to_i,
+      hash["booked_date(2i)"].to_i,
+      hash["booked_date(3i)"].to_i,
+      hash["booked_date(4i)"].to_i,
+      hash["booked_date(5i)"].to_i
+    )
+  end
```

Comment:
Should you also have a params.require().permit()? I wonder if doing that would let you avoid this `date_formatting` method. My guess is the trouble you ran into is that the param is marked as unsafe so rails doesn't naturally want you to dig into it but I am not 100% sure.

Reply:
I can definitely add strong params but somehow I feel that I will always need the date_formating method... The problem is that simple_form return the year, month, day, hour and minutes in 5 different integers (see screenshot below) so the date_formating method is here to reconstitute the date. I think by using this date_formatting method I am already forcing the user to only submit YY-MM-DD-HH-MM (I only use those 5 integers in my method so even if the user maliciously add anotherl field, it won't be taken into account as I am ignoring it in the method) Please let me know if there is a better way :)
![Screen Shot 2022-01-25 at 9 45 42](https://user-images.githubusercontent.com/68413600/150889476-2463c24f-40a4-4e5d-b2b6-a955c12ec340.png)

Reply:
FYI - this is how the simple form looks like, maybe we can modify the input field so that it returns a string and not 5 distinct integers? 
![Screen Shot 2022-01-25 at 9 51 02](https://user-images.githubusercontent.com/68413600/150890000-d1d810da-b13b-4493-9dae-09c7f90ec6d
![Screen Shot 2022-01-25 at 9 51 23](https://user-images.githubusercontent.com/68413600/150890006-ccf9ad88-07c4-42aa-bb84-bb518fba686b.png)
1.png)
)

Reply:
Ah. I see. The way simple_form_for with a date input is different than I expected. I anticipated a date-picker that converts it into a single datetime string. I will have to look more into have simple_form_for works. Thanks for the followup! 🙇

Reply:
`<%= f.input :booked_date, as: :date, html5: true %>`
does that work?

Reply:
I just saw that as well on heartcombo/simple_form github! It is so much better :D Let me modify my code!
![Screen Shot 2022-01-25 at 10 12 17](https://user-images.githubusercontent.com/68413600/150892126-d9002e42-43d4-4a65-85f9-83583f2c6f50.png)
![Screen Shot 2022-01-25 at 10 12 04](https://user-images.githubusercontent.com/68413600/150892135-5bd9c15e-c599-4d9b-9cb6-f173f3ef6376.png)

Reply:
Thank you so much @dostaglou  and @Gakkun89, I committed and pushed a second time!

![Screen Shot 2022-01-25 at 10 36 50](https://user-images.githubusercontent.com/68413600/150894816-e24f8846-0efc-4426-af4c-a711845d87de.png)

Reply:
If you really want to go one extra you can save just the date with `to_date`:
```ruby
[1] pry(main)> DateTime.current
=> Tue, 25 Jan 2022 02:13:07 +0000
[2] pry(main)> DateTime.current.to_date
=> Tue, 25 Jan 2022
```

Reply:
I tried .to_datetime (otherwise the booking is saved with 00:00 as HH:MM) but somehow I still need to convert the params to string for that to work :( Any suggestions?

`booked_date: date_params.to_s.to_datetime`

Reply:
Ah well if your database is saving it as DateTime I suppose there is not much point of calling to_date. The time is not going to be used in the app either way right? But also as Carl mentioned might have to account for difference between UTC and JST